utils/readboundaries_shp.py

```diff
@@ -24,11 +30,37 @@ def quantize(data,least_significant_digit):
     scale = pow(2.,bits)
     return np.around(scale*data)/scale
 
+def interpolate_long_segments(coords, resolution):
+    lookup_thresh = {'c': 0.5, 'l':0.3, 'i':0.2, 'h':0.1, 'f':0.05}
+    thresh = lookup_thresh[resolution]
+    spacing = thresh / 5.0
+
+    lons, lats = coords.T
+    dist = np.hypot(np.diff(lons), np.diff(lats))
+
+    if np.all(dist <= thresh):
+        return coords
+
+    out_lon, out_lat = [], []
+    for i in np.arange(len(dist)):
+        if dist[i] <= thresh:
+            out_lon.append(lons[i])
+            out_lat.append(lats[i])
+        else:
+            x = [0, dist[i]]
+            new_x = np.arange(0, dist[i], spacing)
+            out_lon.extend(np.interp(new_x, x, lons[i:i+2]))
+            out_lat.extend(np.interp(new_x, x, lats[i:i+2]))
```

Comment:
I am wondering if the logic here is faulty. The arange() call will never include `dist[i]` in its output, so then it will have a fairly significant jump to the next spot.